src/Microsoft.PowerShell.Commands.Utility/commands/utility/ReadConsoleCmdlet.cs

```diff
@@ -149,27 +149,37 @@ protected override void BeginProcessing()
                 {
                     foreach (PSObject o in result.Values)
                     {
-                        WriteObject(o);
+                        if (MaskInput && o.BaseObject is SecureString)
+                        {
+                            WriteObject(Utils.GetStringFromSecureString((SecureString)o.BaseObject));
+                        }
```

Comment:
Can still eliminate an unnecessary cast:

```suggestion
                        if (MaskInput && o.BaseObject is SecureString maskedInput)
                        {
                            WriteObject(Utils.GetStringFromSecureString(maskedInput));
                        }
```